### PR TITLE
Refactor to instantiate relations through "newXXX" methods

### LINKED ISSUE #64: Call newHasMany() newHasOne() newBelongsTo() when instantiating
I'd like to use the following two library. Both of them have nice features.

- [staudenmeir/eloquent-eager-limit: Laravel Eloquent eager loading with limit](https://github.com/staudenmeir/eloquent-eager-limit)
- [topclaudy/compoships: Multi-columns relationships for Laravel's Eloquent ORM](https://github.com/topclaudy/compoships)

However, each `Builder` `Relation` is extended from Eloquent core. It's a little hard to go on with both of them.

Would you prefer to refactor them into Trait-based implementation?

Related: staudenmeir/eloquent-eager-limit#19

### PR DESCRIPTION
**Pull request by mpyw**
Resolves #64

This helps extended library to reduce copy-and-paste.

### CODE REVIEW

File: src/Database/Eloquent/Concerns/HasRelationships.php [hunk 1]
```diff
@@ -63,21 +67,38 @@ public function hasOne($related, $foreignKey = null, $localKey = null)
...
+     * Instantiate a new HasOne relationship.
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @param \Illuminate\Database\Eloquent\Model   $parent
+     * @param string|array                          $foreignKey
+     * @param string|array                          $localKey
+     *
+     * @return \Awobaz\Compoships\Database\Eloquent\Relations\HasOne
+     */
+    protected function newHasOne(Builder $query, Model $parent, $foreignKey, $localKey)
```
Comment (cykirsch) [inline src/Database/Eloquent/Concerns/HasRelationships.php]: If these methods are the same as the parent, they shouldn't need to be included here.

Comment (mpyw) [reply src/Database/Eloquent/Concerns/HasRelationships.php]: Reverted 2e22c0f.

Return type does not match to the parent one, they are required.

**Comment by mpyw** [comment]
@topclaudy Thanks! I'll appriciate if the new version is released.

**Comment by topclaudy** [comment]
@mpyw 1.1.15 is released.

**Pull request merged**